qualang_tools/octave_tools/calibration_result_plotter.py

```diff
@@ -0,0 +1,599 @@
+
+        plt.text(
+            np.min(fine_q_scan) + 0.5 * width,
+            np.min(fine_i_scan) + 0.5 * height,
+            f"DC phase={d.corrections.dc_phase:.5f}\nDC gain={d.corrections.dc_gain:.5f}",
+            color="#ff7f0e",
+            verticalalignment="bottom",
+            horizontalalignment="left",
+        )
+
+        plt.plot(x0, y0, "yo", markersize=8)
+        plt.plot(x0, y0, "ro", markersize=4)
+
+        t = plt.text(
+            np.min(fine_q_scan) + 0.5 * width,
+            np.max(fine_i_scan) - 0.5 * height,
+            f"fine scan\nLO = {self.lo_frequency / 1e9:.3f}GHz",
+            color="k",
+            verticalalignment="top",
+        )
+        t.set_bbox(dict(facecolor="w", alpha=0.8))
+
+        plt.text(
+            x0,
+            y0 - height * 2,
+            f"Q_dc={x0:.2f}mV\nI_dc={y0:.2f}mV",
+            color="y",
+            horizontalalignment="center",
+            verticalalignment="top",
+        )
+
+        plt.subplot(223)
+
+        X, Y = d.q_scan, d.i_scan
+        p = d.fit.pol
+
+        iq_error = self.u.demod2volts(
+            p[0] + p[1] * X + p[2] * Y + p[3] * X**2 + p[4] * X * Y + p[5] * Y**2 - d.lo, integration_length
+        )
+        iq_error_dbm = self._convert_to_dbm(np.abs(iq_error))
+
+        plt.pcolor(
+            fine_q_scan,
+            fine_i_scan,
+            iq_error_dbm,
+            cmap=CalibrationResultPlotter.custom_cmap,
+        )
+        plt.xlabel("Q (mV)")
+        plt.ylabel("I (mV)")
+        plt.axis("equal")
+
+        t = plt.text(
+            np.min(fine_q_scan) + 0.5 * width,
+            np.max(fine_i_scan) - 1.5 * height,
+            "fit error",
+            color="k",
+        )
+        t.set_bbox(dict(facecolor="w", alpha=0.8))
+
+        plt.colorbar(label="Power [dBm]")
+
+        plt.subplot(221)
+
+        plt.xlim(0, 1)
+        plt.ylim(0, 1)
+
+        content = [
+            "Current result:",
+            f"      I_dc = {y0:.02f}mV, Q_dc = {x0:.2f}mV",
+            "\nAchieved LO supression:",
+            f"{self.get_lo_leakage_rejection():.3f} dB",
+        ]
+
+        plt.text(
+            0.05,
+            0.75,
+            "\n".join(content),
+            horizontalalignment="left",
+            verticalalignment="top",
+            bbox=dict(facecolor="none", edgecolor="black", boxstyle="round,pad=1"),
+        )
+        plt.box(False)
+        plt.xticks([])
+        plt.yticks([])
+        plt.suptitle(f"LO auto calibration @ {self.lo_frequency/1e9:.3f}GHz")
+        plt.tight_layout()
+
+    def show_image_rejection_calibration_result(
+        self,
+    ) -> None:
+        """
+        Plots the results of the IF image calibration process.
+
+        This method generates a series of subplots that visualize the coarse and fine scans of the IF image calibration
+        and returns the achieved image suppression in dB.
```

Comment:
I would also specify that the displayed powers correspond to the powers measured by the OPX after up and down conversion and may differ significantly from the power outputted by the Octave, or something along these lines
A well as the meaning of dB here "Here the returned value is with respect to before the calibration (so taking into account the previously load calibration parameters), or without any parameters?"